independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/SwitchToEventLoopHandler.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.resteasy.reactive.server.handlers;
+
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+import org.jboss.resteasy.reactive.server.core.BlockingOperationSupport;
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+/**
+ * A special handler whose only purpose is to switch execution from a worker thread to the event loop
+ * This can be needed when for example a large payload is read but the JAX-RS Resource method
+ * is intended to be run on the event loop.
+ */
+public class SwitchToEventLoopHandler implements ServerRestHandler {
+
+    private final Supplier<Executor> workerExecutorSupplier;
+
+    public SwitchToEventLoopHandler(Supplier<Executor> workerExecutorSupplier) {
+        this.workerExecutorSupplier = workerExecutorSupplier;
+    }
+
+    @Override
+    public void handle(ResteasyReactiveRequestContext requestContext) throws Exception {
```

Comment:
Hi, since I also looked at the bug for a bit I am of course interested in how it is fixed. I hope it's fine if I ask a question as someone who isn't familiar with the codebase:

What are the differences between this handler and the `NonBlockingHandler`? They seem to have similar goals (but it is unclear to me where their goals differ), but they differ in various details, such as `BlockingOperationSupport.isBlockingAllowed()` vs. `!requestContext.serverRequest().isOnIoThread()`, `requestContext.resumeOnEventLoop()` vs. `requestContext.resume(requestContext.getContextExecutor())` and executing the previous code on a worker executor vs. directly running it on the current thread.

If you happen to have a few minutes, I would really appreciate a quick explanation of these choices. Thanks!